File: shoring/fxapp/src/main/java/com/panopset/fxapp/PanApplication.java
package com.panopset.fxapp;

public interface PanApplication {

  int DEFAULT_WIDTH = 1000;
  int DEFAULT_HEIGHT = 900;

  default int getDefaultWidth() {
    return DEFAULT_WIDTH;
  }

  default int getDefaultHeight() {
    return DEFAULT_HEIGHT;
  }
  
  String getCompanyName();

  String getApplicationDisplayName();

  String getDescription();
  
  default String getApplicationShortName() {
    return this.getClass().getName().toLowerCase();
  }

  default String getFilesKey() {
    return getApplicationShortName() + "_files";
  }
}

File: code/shoring/fxapp/src/main/kotlin/com/panopset/fxapp/PanApplication.kt
package com.panopset.fxapp

import java.util.*

abstract class PanApplication(private val applicationInfo: ApplicationInfo) {

    fun getCompanyName(): String {
        return applicationInfo.getApplicationBranding().getCompanyName()
    }
    fun getApplicationDisplayName(): String {
        return applicationInfo.getApplicationDisplayName()
    }
    fun getDescription(): String {
        return applicationInfo.getDescription()
    }

    open fun doAfterShow(fxDoc: FxDoc) {

    }

    val applicationShortName = this.javaClass.name.lowercase(Locale.getDefault())
    val filesKey = applicationShortName + "_files"

    var id = 0L
    fun getNextUniqueID(): Long {
        return id++
    }
}
